lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -93,54 +93,81 @@ public Thread StartTaskOnNewThread()
 
             var taskThread = new Thread(() =>
             {
+                // The result of the task execution.
+                byte[] resultReturnedByTask = null;
+
+                // Whether or not a result shall be returned to the Driver.
+                bool returnResultToDriver = true;
+
+                // Exception thrown during `Dispose`, if any.
+                Exception exceptionThrownByTaskDispose = null;
+
                 try
                 {
+                    // Run the handlers for `TaskStart`
                     Logger.Log(Level.Verbose, "Set running status for task");
+                    _currentStatus.RunTaskStartHandlers();
```

Comment:
The original _taskLifeCycle.Start(0 was in the _heartBeatManager lock. Do you on purposely take it out? Any reason?